ironfish/src/rpc/routes/chain/getTransaction.ts

```diff
@@ -101,80 +83,66 @@ router.register<typeof GetTransactionRequestSchema, GetTransactionResponse>(
       throw new ValidationError(`Missing transaction hash`)
     }
 
-    const hashBuffer = request.data.blockHash
+    const transactionHashBuffer = Buffer.from(request.data.transactionHash, 'hex')
+
+    const blockHashBuffer = request.data.blockHash
       ? BlockHashSerdeInstance.deserialize(request.data.blockHash)
-      : await node.chain.getBlockHashByTransactionHash(
-          Buffer.from(request.data.transactionHash, 'hex'),
-        )
+      : await node.chain.getBlockHashByTransactionHash(transactionHashBuffer)
 
-    if (!hashBuffer) {
+    if (!blockHashBuffer) {
       throw new ValidationError(
         `No block hash found for transaction hash ${request.data.transactionHash}`,
       )
     }
 
-    const blockHeader = await node.chain.getHeader(hashBuffer)
+    const blockHeader = await node.chain.getHeader(blockHashBuffer)
     if (!blockHeader) {
-      throw new ValidationError(`No block found`)
+      throw new ValidationError(
+        `No block found for block hash ${blockHashBuffer.toString('hex')}`,
+      )
     }
 
-    // Empty response used for case that transaction not found
-    const rawTransaction: GetTransactionResponse = {
-      fee: '0',
-      expiration: 0,
-      noteSize: 0,
-      notesCount: 0,
-      spendsCount: 0,
-      signature: '',
-      notesEncrypted: [],
-      spends: [],
-      notes: [],
-      mints: [],
-      burns: [],
-    }
     const transactions = await node.chain.getBlockTransactions(blockHeader)
 
-    transactions.map(({ transaction, initialNoteIndex }) => {
-      if (transaction.hash().toString('hex') === request.data.transactionHash) {
-        const fee = transaction.fee().toString()
-        const expiration = transaction.expiration()
-        const signature = transaction.transactionSignature()
-        const notesEncrypted = []
-
-        for (const note of transaction.notes) {
-          notesEncrypted.push(note.serialize().toString('hex'))
-        }
+    const foundTransaction = transactions.find(({ transaction }) =>
+      transaction.hash().equals(transactionHashBuffer),
+    )
 
-        rawTransaction.fee = fee
-        rawTransaction.expiration = expiration
-        rawTransaction.noteSize = initialNoteIndex + transaction.notes.length
-        rawTransaction.notesCount = transaction.notes.length
-        rawTransaction.spendsCount = transaction.spends.length
-        rawTransaction.signature = signature.toString('hex')
-        rawTransaction.notesEncrypted = notesEncrypted
-
-        rawTransaction.spends = transaction.spends.map((spend) => ({
-          nullifier: spend.nullifier.toString('hex'),
-          commitment: spend.commitment.toString('hex'),
-          size: spend.size,
-        }))
-
-        rawTransaction.notes = transaction.notes.map((note) => ({
-          hash: note.hash().toString('hex'),
-          serialized: note.serialize().toString('hex'),
-        }))
+    if (!foundTransaction) {
+      throw new ValidationError(
```

Comment:
NotFound error fits better than Validation. In general  Validation error (INVALID_ARGUMENT) indicates arguments that are problematic regardless of the state of the system (e.g., a malformed file name).